pkg/operator/sync.go

```diff
@@ -341,26 +343,91 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		caData[CA.File] = string(CA.Data)
 	}
 
-	err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Delete(context.TODO(), "merged-trusted-image-registry-ca", metav1.DeleteOptions{})
+	cm, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Get(context.TODO(), "merged-trusted-image-registry-ca", metav1.GetOptions{})
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
-
 	cmAnnotations := make(map[string]string)
 	cmAnnotations["openshift.io/description"] = "Created and managed by the machine-config-operator"
-	_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Create(
-		context.TODO(),
-		&corev1.ConfigMap{
+	if err != nil && errors.IsNotFound(err) {
+		klog.Infof("creating merged-trusted-image-registry-ca")
+		_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Create(
+			context.TODO(),
+			&corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "merged-trusted-image-registry-ca",
+					Annotations: cmAnnotations,
+				},
+				Data: caData,
+			},
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		klog.Info("merged-trusted-image-registry-ca still exists, checking contents")
+		cmMarshal, err := json.Marshal(cm)
+		if err != nil {
+			return err
+		}
+		newCM := &corev1.ConfigMap{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:        "merged-trusted-image-registry-ca",
 				Annotations: cmAnnotations,
 			},
 			Data: caData,
-		},
-		metav1.CreateOptions{},
-	)
-	if err != nil {
-		return err
+		}
+		newCMMarshal, err := json.Marshal(newCM)
+		if err != nil {
+			return err
+		}
+
+		// check for deletedData
+		stillExists := false
+		for file, data := range cm.Data {
+			stillExists = false
+			// for each file in the configmap
+			// does it match ANY of the new files? if not, it is deleted.
+			for newfile, newdata := range caData {
+				if newfile == file && newdata == data {
+					stillExists = true
+				}
+			}
+			if !stillExists {
+				break
+			}
+		}
+		if !stillExists && len(cm.Data) > 0 {
+			klog.Info("Detecting deletions in merged-trusted-image-registry-ca, updating")
+			_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Update(context.TODO(), newCM, metav1.UpdateOptions{})
```

Comment:
Question: just curious, why is this an update operation (when a sub part of the ca is deleted) and the below additions are patches? Would it be possible to batch both into one operation since you're doing a 3 way patch below regardless? (not necessary, just curious)

Reply:
Not 100% necessary if that breaks things, we can switch it back as well.

Reply:
seems to work